<launch>
  <!-- hostname of your notebook -->
  <arg name="notebook" default="sfcps" />
  <!-- default hostname of the robot -->
  <arg name="robot" default="daisy" />
  <!-- ROS distro that should be initialized on the rover -->
  <arg name="robot-distro" default="indigo" />
  <!-- robot's interface to arcos (firmware/driver for motors) -->
  <arg name="robot-port" default="/dev/ttyTHS1" />
  <!-- default hostname of the robot's Raspberry Pi -->
  <arg name="pi" default="daisy-pi" />

  <!-- machines -->
  <machine name="notebook" address="$(arg notebook)" default="true" />
  <machine name="robot" address="$(arg robot)"
       env-loader="/opt/ros/$(arg robot-distro)/env.sh" />
  <machine name="pi" address="$(arg pi)"
       env-loader="/opt/ros/kinetic/env_pi.bash" />

  <remap from="/p2os/cmd_vel" to="/teleop/cmd_vel" />

  <!-- nodes running on the robot -->
  <node machine="robot" ns="p2os" name="p2os_driver"
    pkg="p2os_driver" type="p2os_driver"
    output="screen">
    <param name="port" value="$(arg robot-port)" /> <!-- interface to arcos -->
    <param name="pulse" value="1.0" /> <!-- turn annoying beep of -->
    <param name="use_sonar" value="false" /> <!-- sonar sensors off -->
  </node>

  <!-- nodes running on the raspberry pi -->
  <!-- gyroscope -->
  <node machine="pi" ns="pi" name="imu3000"
        pkg="imu3000" type="imu3000"
        output="screen" />
  <!-- accelerometer -->
  <node machine="pi" ns="pi" name="kxtf9"
        pkg="kxtf9" type="kxtf9"
        output="screen" />

  <!-- nodes running on the notebook -->
  <node name="teleop" pkg="pioneer_teleop" type="keyboard" output="screen" />
  <node name="pose_estimator" pkg="pose_visualization" type="pose_estimator.py" output="screen" />
  <node name="coordinatetransforms" pkg="pose_visualization" type="coordinatetransforms.py" output="screen" />
  <node name="odomtransform" pkg="tf" type="static_transform_publisher" output="screen" args="0 0 0 0 0 0 estimate odom 10"/>


  <!-- enable motors (this node will quit after 3s) -->
  <node name="rostopic_pub_enable_motors"
    pkg="rostopic" type="rostopic"
    args="pub /p2os/cmd_motor_state p2os_msgs/MotorState 1 -1" />

</launch>
